[Bug] Docker build 构建失败

### 💻 Operating System

Ubuntu

### 📦 Environment

Docker

### 🌐 Browser

Chrome

### 🐛 Bug Description

使用以下命令构建失败

```
git clone https://github.com/lobehub/lobe-chat.git
cd lobe-chat/
docker build -t lobe-chat:v0.128.5 .
```

报错日志如下

```
=> [sharp 1/3] RUN corepack enable                                                                                                                                                                            0.8s
 => [runner 1/9] WORKDIR /app                                                                                                                                                                                  0.8s
 => [runner 2/9] RUN addgroup --system --gid 1001 nodejs                                                                                                                                                       0.7s
 => [sharp 2/3] WORKDIR /app                                                                                                                                                                                   0.0s
 => [builder 3/6] COPY package.json ./                                                                                                                                                                         0.0s
 => [sharp 3/3] RUN pnpm add sharp                                                                                                                                                                            13.3s
 => ERROR [builder 4/6] RUN pnpm i                                                                                                                                                                            30.9s
 => [runner 3/9] RUN adduser --system --uid 1001 nextjs                                                                                                                                                        0.5s
------
 > [builder 4/6] RUN pnpm i:
6.898  WARN  GET https://registry.npmjs.org/@testing-library%2Freact error (ECONNRESET). Will retry in 10 seconds. 2 retries left.
7.569 Progress: resolved 1, reused 0, downloaded 0, added 0
8.571 Progress: resolved 15, reused 0, downloaded 9, added 0
9.574 Progress: resolved 41, reused 0, downloaded 29, added 0
10.74 Progress: resolved 56, reused 0, downloaded 46, added 0
11.75 Progress: resolved 73, reused 0, downloaded 59, added 0
12.78 Progress: resolved 85, reused 0, downloaded 72, added 0
13.80 Progress: resolved 94, reused 0, downloaded 78, added 0
15.37 Progress: resolved 111, reused 0, downloaded 98, added 0
16.37 Progress: resolved 117, reused 0, downloaded 114, added 0
17.37 Progress: resolved 119, reused 0, downloaded 116, added 0
18.37 Progress: resolved 151, reused 0, downloaded 143, added 0
19.37 Progress: resolved 171, reused 0, downloaded 157, added 0
20.37 Progress: resolved 209, reused 0, downloaded 195, added 0
21.38 Progress: resolved 244, reused 0, downloaded 227, added 0
22.38 Progress: resolved 287, reused 0, downloaded 263, added 0
23.38 Progress: resolved 324, reused 0, downloaded 306, added 0
24.38 Progress: resolved 364, reused 0, downloaded 341, added 0
25.38 Progress: resolved 404, reused 0, downloaded 377, added 0
26.38 Progress: resolved 436, reused 0, downloaded 405, added 0
27.38 Progress: resolved 467, reused 0, downloaded 437, added 0
28.38 Progress: resolved 500, reused 0, downloaded 463, added 0
29.39 Progress: resolved 530, reused 0, downloaded 498, added 0
30.33  ERR_PNPM_WORKSPACE_PKG_NOT_FOUND  In : "@auth/core@workspace:*" is in the dependencies but no package named "@auth/core" is present in the workspace
30.33
30.33 This error happened while installing the dependencies of next-auth@5.0.0-beta.12
30.33
30.33 Packages found in the workspace:
------
Dockerfile:25
--------------------
  23 |     # If you want to build docker in China
  24 |     # RUN npm config set registry https://registry.npmmirror.com/
  25 | >>> RUN pnpm i
  26 |
  27 |     COPY . .
--------------------
ERROR: failed to solve: process "/bin/sh -c pnpm i" did not complete successfully: exit code: 1
```

### 🚦 Expected Behavior

_No response_

### 📷 Recurrence Steps

_No response_

### 📝 Additional Information

_No response_

--- a/docs/self-hosting/environment-variables/basic.mdx
+++ b/docs/self-hosting/environment-variables/basic.mdx
@@ -71,6 +71,13 @@ Further reading:
 - Default: `-`
 - Example: `Tfhi2t2pelSMEA8eaV61KaqPNEndFFdMIxDaJnS1CUI=`
 
+#### `NEXTAUTH_URL`
+
+- Type: Optional
+- Description: This URL is used to specify the callback address for Auth.js during OAuth authentication. It does not need to be set when deploying on Vercel.
+- Default: `-`
+- Example: `https://example.com/api/auth`
+
 ### Auth0
 
 <Callout>

--- a/docs/self-hosting/environment-variables/basic.zh-CN.mdx
+++ b/docs/self-hosting/environment-variables/basic.zh-CN.mdx
@@ -71,6 +71,13 @@ LobeChat 在部署时提供了一些额外的配置项，你可以使用环境�
 - 默认值: `-`
 - 示例: `Tfhi2t2pelSMEA8eaV61KaqPNEndFFdMIxDaJnS1CUI=`
 
+#### `NEXTAUTH_URL`
+
+- 类型：可选
+- 描述：该URL用于指定Auth.js在执行OAuth验证时的回调地址，在Vercel上部署时无需设置。
+- 默认值：`-`
+- 示例：`https://example.com/api/auth`
+
 ### Auth0
 
 <Callout>

--- a/package.json
+++ b/package.json
@@ -106,7 +106,7 @@
     "modern-screenshot": "^4",
     "nanoid": "^5",
     "next": "^14.1",
-    "next-auth": "beta",
+    "next-auth": "5.0.0-beta.11",
     "numeral": "^2.0.6",
     "nuqs": "^1.15.4",
     "openai": "^4.22",

--- a/src/app/api/auth/next-auth.ts
+++ b/src/app/api/auth/next-auth.ts
@@ -17,6 +17,7 @@ const nextAuth = NextAuth({
       ]
     : [],
   secret: NEXTAUTH_SECRET,
+  trustHost: true,
 });
 
 export const {

--- a/src/middleware.ts
+++ b/src/middleware.ts
@@ -13,7 +13,10 @@ const defaultMiddleware = () => NextResponse.next();
 const withAuthMiddleware = auth((req) => {
   // Just check if session exists
   const session = req.auth;
-  const isLoggedIn = !!session;
+
+  // Check if next-auth throws errors
+  // refs: https://github.com/lobehub/lobe-chat/pull/1323
+  const isLoggedIn = !!session?.expires;
 
   // Remove & amend OAuth authorized header
   const requestHeaders = new Headers(req.headers);

